### PR TITLE
fix: 画布中右键弹框，修复添加父级-弹出框功能

### PR DESCRIPTION
**Pull request by lhuans**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

弹出框右键功能无效

## What is the new behavior?
右键选择弹出框
image
会出现如下效果
image
出码结果如下
image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
LGTM

File: packages/canvas/src/components/container/CanvasMenu.vue [hunk 1]
```diff
@@ -132,19 +132,62 @@ export default {
...
-            children: [schema]
-          }
 
-          parent.children.splice(index, 1, wrapSchema)
+        if (!schema || !parent) {
+          return
+        }
 
-          getController().addHistory()
+        const index = parent.children.indexOf(schema)
```
Comment (chilingling) [inline packages/canvas/src/components/container/CanvasMenu.vue]: 这里换成 `findIndexOf` 会不会更好？`parent.children.findIndexOf(({ id }) => id === schema.id)` 类似这样的

File: packages/canvas/src/components/container/CanvasMenu.vue [hunk 2]
```diff
@@ -132,19 +132,62 @@ export default {
...
+            },
+            children: [
+              {
+                componentName: 'Template',
+                props: {
+                  slot: 'reference'
+                },
+                children: [
+                  {
+                    componentName: 'div',
```
Comment (chilingling) [inline packages/canvas/src/components/container/CanvasMenu.vue]: 这里 referrence 不套一层 div 可以吗？用户可能不想要多一层 div ？

Comment (lhuans) [reply packages/canvas/src/components/container/CanvasMenu.vue]: > 这里 referrence 不套一层 div 可以吗？用户可能不想要多一层 div ？

OK，已修改

File: packages/canvas/src/components/container/CanvasMenu.vue (hunk 1 above)
Comment (lhuans) [reply packages/canvas/src/components/container/CanvasMenu.vue]: > 这里换成 `findIndexOf` 会不会更好？`parent.children.findIndexOf(({ id }) => id === schema.id)` 类似这样的

已修改

**Review by chilingling — APPROVED** [review]
LGTM

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**